internal/cmdrender/runtime/container.go

```diff
@@ -58,27 +89,38 @@ type ContainerFn struct {
 // It reads the input from the given reader and writes the output
 // to the provided writer.
 func (f *ContainerFn) Run(reader io.Reader, writer io.Writer) error {
-	pr := printer.FromContextOrDie(f.Ctx)
+	// check and pull image before running to avoid polluting CLI
+	// output
+	err := f.prepareImage()
+	if err != nil {
+		return fmt.Errorf("failed to check function existence: %w", err)
```

Comment:
Not this PR, but we will have to add an internal-error-type to represent these errors because user won't be able to understand what this means. This is sort of internal failure.